### PR TITLE
Smart Plug UI

### PR DESCRIPTION
**Pull request by benfrancis**
Smart plug UI.

### CODE REVIEW

File: static/js/things.js [hunk 1]
```diff
@@ -182,6 +187,11 @@ var ThingsScreen = {
             console.log('rendering new multi level switch');
             thing = new MultiLevelSwitch(description, 'htmlDetail');
             break;
+          case 'smartPlug':
+            console.log('rendering new smart plug');
+            // eslint-disable-next-line no-unused-vars
+            var newSmartPlug = new SmartPlug(description, 'htmlDetail');
```
Comment (hobinjk) [inline static/js/things.js]: This should instead by `thing = new SmartPlug...`

File: static/js/smart-plug.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+ * Show updated power consumption value.
+ */
+SmartPlug.prototype.updatePower = function(power) {
+  this.properties.power = power;
+  if (!this.properties.on) {
+    this.smartPlugLabel.innerText = 'off';
+  } else if (power) {
+    this.smartPlugLabel.innerText = power + 'W';
+  } else {
+    this.smartPlugLabel.innerText = '...';
```
Comment (hobinjk) [inline static/js/smart-plug.js]: I think it might be a bit clearer here to have 'on' because '...' is used by onOffSwitch to mean API failure

**Comment by benfrancis** [comment]
Thanks :)

**Comment by hobinjk** [comment]
@benfrancis Can you follow-up with a change to address the second comment (`var newSmartPlug = new SmartPlug(description, 'htmlDetail');` -> `thing = new SmartPlug(...)` in things.js)? It's a small fix but it's necessary to make the title and icon show up

**Comment by hobinjk** [comment]
Oh wait, looks like GitHub's diff is just out of date, don't mind me

**Pull request merged**